### DISCUSSION TITLE
how to get multiple rows into one panel using template?

### DISCUSSION THREAD

**Discussion by ikke-t** [flant/grafana-statusmap#190] [Q&A]
Hi,

I'm newbie doing templates. I like how this panel looks like, and I'd like to get all my HAproxy backends statuses into nice panel. However, I don't know grafana json enough to place them into one panel. I get them all into separate panels using the vertical repeat, but how to get them into a single panel?

Here's how I get them stacked into separate panels:

```
 "panels": [
    {
      "collapsed": false,
      "datasource": null,
      "gridPos": {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "panels": [],
      "repeat": null,
      "title": "backend statuses",
      "type": "row"
    },
    {
      "cards": {
        "cardHSpacing": 2,
        "cardMinWidth": 5,
        "cardRound": null,
        "cardVSpacing": 2
      },
      "color": {
        "cardColor": "#b4ff00",
        "colorScale": "sqrt",
        "colorScheme": "interpolateGnYlRd",
        "defaultColor": "#757575",
        "exponent": 0.5,
        "mode": "discrete",
        "thresholds": [
          {
            "color": "#C4162A",
            "value": "0",
            "tooltip": "offline"
          },
          {
            "color": "#37872D",
            "value": "1",
            "tooltip": "online"
          }
        ],
        "min": 0,
        "max": 1
      },
      "datasource": "telegraf",
      "gridPos": {
        "h": 4,
        "w": 24,
        "x": 0,
        "y": 1
      },
      "highlightCards": true,
      "id": 3,
      "legend": {
        "show": true
      },
      "nullPointMode": "as empty",
      "options": {},
      "pageSize": 15,
      "seriesFilterIndex": -1,
      "statusmap": {
        "ConfigVersion": "v1"
      },
      "targets": [
        {
          "measurement": "haproxy",
          "orderByTime": "ASC",
          "rawQuery": true,
          "refId": "A",
          "resultFormat": "time_series",
          "select": [
            [
              {
                "params": [
                  "active_servers"
                ],
                "type": "field"
              },
              {
                "params": [],
                "type": "min"
              }
            ]
          ],
          "tags": [
            {
              "key": "host",
              "operator": "=~",
              "value": "/^$host$/"
            },
            {
              "condition": "AND",
              "key": "proxy",
              "operator": "=~",
              "value": "/^$proxy$/"
            }
          ],
          "groupBy": [
            {
              "type": "time",
              "params": [
                "$__interval"
              ]
            },
            {
              "type": "fill",
              "params": [
                "none"
              ]
            }
          ],
          "alias": "$proxy",
          "query": "SELECT min(\"active_servers\") FROM \"haproxy\" WHERE (\"host\" =~ /^$host$/ AND \"proxy\" =~ /^$proxy$/) AND $timeFilter GROUP BY time($__interval) fill(none)\n"
        }
      ],
      "timeFrom": null,
      "timeShift": null,
      "title": "Backend status",
      "tooltip": {
        "freezeOnClick": true,
        "items": [],
        "show": true,
        "showItems": false
      },
      "type": "flant-statusmap-panel",
      "useMax": true,
      "usingPagination": false,
      "xAxis": {
        "show": true
      },
      "yAxis": {
        "maxWidth": -1,
        "minWidth": -1,
        "show": true
      },
      "yAxisSort": "metrics",
      "repeat": "proxy",
      "repeatDirection": "v"
    },
```

![Screenshot from 2020-11-02 22-45-51](https://user-images.githubusercontent.com/1146409/97917590-59c24500-1d5d-11eb-8875-b1b3aa85b3f4.png)

According to [the instructions](https://grafana.com/grafana/plugins/flant-statusmap-panel), I should have multiple queries. How to achieve that using the $proxy variable?

![multiple queries](https://raw.githubusercontent.com/flant/grafana-statusmap/master/src/img/queries-example.png)

I'd like to have them all in one panel like here:
![multiple rows](https://raw.githubusercontent.com/flant/grafana-statusmap/master/src/img/flant-statusmap-panel-dark.png)

**Comment by diafour** [comment]
Hello! You want a single panel, so stacking is not needed. You need to add a standalone panel with a query that accepts multiple values stored in template variable `$proxy`. Use a suitable field or tag for alias and the plugin will show multiple rows. I've found this interesting page https://grafana.com/docs/grafana/latest/variables/formatting-multi-value-variables/ Maybe it helps.

**Comment by ikke-t** [comment]
thanks, that page suggests to use format ```(host1|host2|host3)```, but I don't get how to make it pick up all selections from ```/^$proxy$/``` variable?

**Comment by diafour** [comment]
Just leave it as is. `/^$proxy$/` become `/^(host1|host2|host3)$/` — that is a valid regex.

**Comment by ikke-t** [comment]
No, unfortunately it won't work so. It does not generate multiple queries into single panel. If I have it like it is, but not having the repeat, it does this (with two  selected as example):

![Screenshot from 2020-11-03 21-32-48](https://user-images.githubusercontent.com/1146409/98032110-2e525f80-1e1c-11eb-9195-edf10396abf9.png)

So it joins them into single query, not creating several query lines. After googling a lot after this issue, I  start to feel there is no way to do such table as the coffee machine example using influxdb tag variables with this panel plugin.

**Comment by ikke-t** [comment]
where as if I manually would create all tags, I'd get this look what I'm after, single panel with multiple lines:

![Screenshot from 2020-11-03 21-42-01](https://user-images.githubusercontent.com/1146409/98033015-889ff000-1e1d-11eb-8e01-c4ddfc35ce50.png)

**Comment by ikke-t** [comment]
so I had to hardcode it like this:

![Screenshot from 2020-11-03 21-45-19](https://user-images.githubusercontent.com/1146409/98033274-f51aef00-1e1d-11eb-811d-2636aeb0f963.png)

**Comment by diafour** [comment]
It looks that $proxy as an Alias is just a value of template variable. I don't have experience with a Flux language. May be `proxy` should be in a GROUP BY or in a SELECT, so you can use its value in an Alias instead of `$proxy` from template variable? Also, it can be just a name overlapping.
